### PR TITLE
Refactor converter

### PR DESCRIPTION
**Pull request by blootsvoets**
Refactor of the converter/processor setup + logging
Idea is that a ConverterFactory can be used without context, as well as a FileProcessorFactory. The Converter and FileProcessor are fully initialised. The latter can also build up some record-specific context.

Also moved some classes to class-specific files.

### CODE REVIEW

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt [hunk 1]
```diff
@@ -105,7 +90,7 @@ class UploadSourceTask : SourceTask() {
 
         logger.info("Polling new records...")
         val records: List<RecordDTO> = try {
-            uploadClient.pollRecords(PollDTO(1, converters.map { it.sourceType }.toSet())).records
+            uploadClient.pollRecords(PollDTO(1, converters.keys)).records
```
Comment (nivemaham) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt]: Aren't the keys here classnames of the converter, whereas I remember the REST resource, fetches based on source-type name?

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/LogRepository.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+interface LogRepository {
+    fun recordLogger(logger: Logger, recordId: Long): RecordLogger
+    fun recordLogger(clazz: Class<*>, recordId: Long): RecordLogger
+
+    val recordIds: Set<Long>
+    fun extract(recordId: Long, reset: Boolean = false): Log?
+}
+
+interface RecordLogger {
```
Comment (nivemaham) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/LogRepository.kt]: Maybe it is better to add documentation on what each of these interfaces mean? I am little lost when I see the method `recordLogger` and with an interface`RecordLogger`. when we call method recordLogger do we "record" a log, or we create an instance of `RecordLogger`?

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/RecordConverter.kt [hunk 3]
```diff
@@ -96,19 +70,33 @@ abstract class RecordConverter(override val sourceType: String, private val avro
...
+            recordLogger.error("Temporarily could not convert record $recordId", exe)
+            throw ConversionTemporarilyFailedException("Temporarily could not convert record $recordId", exe)
         }
     }
 
-    /** process file content with the record data. The implementing method should close response-body. */
-    abstract fun processData(contents: ContentsDTO, inputStream: InputStream, record: RecordDTO, timeReceived: Double)
-            : List<TopicData>
+    override fun convertFile(record: RecordDTO, contents: ContentsDTO, inputStream: InputStream, recordLogger: RecordLogger): List<FileProcessorFactory.TopicData> {
+        val processorFactory = processorFactories.firstOrNull { it.matches(contents) }
```
Comment (nivemaham) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/RecordConverter.kt]: how do we match a FileProcessorFactory based on ContentsDTO?

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt (hunk 1 above)
Comment (blootsvoets) [reply kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt]: This was a change in the `createConverter`: it returns a pair sourceType to converter. I've made this more explicit in the current class now.

**Pull request merged**